plugins/chat/assets/javascripts/discourse/routes/chat-channel.js

```diff
@@ -9,14 +9,18 @@ export default class ChatChannelRoute extends DiscourseRoute {
   @service chatStateManager;
 
   @action
-  willTransition(transition) {
-    this.chat.activeChannel.activeThread = null;
+  willTransition() {
+    if (this.chat.activeChannel) {
+      this.chat.activeChannel.activeThread = null;
+    }
+
     this.chatStateManager.closeSidePanel();
+  }
 
-    if (!transition?.to?.name?.startsWith("chat.")) {
-      this.chatStateManager.storeChatURL();
-      this.chat.activeChannel = null;
-      this.chat.updatePresence();
-    }
+  @action
+  deactivate() {
+    this.chatStateManager.storeChatURL();
+    this.chat.activeChannel = null;
+    this.chat.updatePresence();
```

Comment:
I had done this in the past but had issues with it, would be very if can do it now